### PR TITLE
feat(observability): add PayoutTransactionsAmountInvariant

### PR DESCRIPTION
**Pull request by frankie567**
Add new invariant rule that validates payout transaction amounts correspond
to the sum of transactions that reference them via payout_transaction_id.

- PayoutTransactionsAmountInvariant: checks if payout amounts match sum
  of paid transactions
- Excludes reversed payouts (by design, when reversed payout_transaction_id
  is reset to NULL on paid transactions)
- Uses abs(diff) > 0 to catch mismatches in both directions
- Follows existing invariant pattern with LIMIT=10 and has_more flag

Tests cover:

- Success cases (no payouts, matching amounts, zero amount)
- Failure cases (amount mismatch, missing transactions, sum exceeds payout,
  over limit)
- Edge case (reversed payouts excluded)

<a href="https://cubic.dev/pr/polarsource/polar/pull/13116?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    LIMIT = 10
+
+    async def check(self) -> None:
+        # Payout transactions without reversal (same payout_id)
+        payout_subq = (
+            select(Transaction.id, Transaction.amount)
+            .where(
+                Transaction.type == TransactionType.payout,
+                ~Transaction.payout_id.in_(
```
[the remark this answers is not in the document]
Comment (frankie567) [reply server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py:57]: @cubic-dev-ai Payout transactions _always_ have a Payout associated, so this is a nitpick.

File: server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            .where(Transaction.payout_transaction_id.isnot(None))
+            .group_by(Transaction.payout_transaction_id)
+            .subquery()
+        )
+
+        # Find mismatches
+        diff_expr = func.abs(payout_subq.c.amount) - func.coalesce(
+            paid_subq.c.total_amount, 0
+        )
+        statement = (
```
Comment (psincraian) [inline server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py:82]: Should we limit the data to the last X days only? I mean we are mainly interested in new payouts so I don't see the benefit of checking older ones than 30d for example

**Review by psincraian — APPROVED** [review]

File: server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py (hunk 2 above)
Comment (frankie567) [reply server/polar/observability/invariants/rules/payout_transactions_amount_invariant.py:82]: Hmm, true, but I'm concerned we may have some bugs/issues that deteriorate existing data. But that may be a bit paranoid.
